### PR TITLE
Fixed #33564 -- Confirmed support for PROJ 9.X.

### PR DESCRIPTION
**Pull request by smithdc1**

### CODE REVIEW

**Comment by smithdc1** [comment]
GEOS has also now got a `CMake` build which is noted in their docs and works for me with their latest release. I've therefore push a separate commit incase you didn't want to take that change at this time. 

https://github.com/libgeos/geos/blob/main/INSTALL.md

GDAL is also moving to `CMake` but those changes have not yet made it into a release. https://github.com/OSGeo/gdal/pull/4663

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1: The first commit should be a part of confirming support for PROJ 9 as we've done in the past, see 065832eaec167a45008aa125887ce1215a1f257d. Can you confirm/add support for PROJ 9?

**Comment by smithdc1** [comment]
I can't confirm support for PROJ 9x. 

My usual technique of using conda to provide an isolated environment with all the packages built isn't working as yet. Not all of the packages have been updated to declare PROJ 9x support.

I was working my way through trying to build everything my self but have only got to these two packages as yet and I'm not so confident I'll be able to make it work.

**Comment by felixxm** [comment]
> I can't confirm support for PROJ 9x.

np, I will add this to my TODO list :wink: and incorporate your changes.

**Comment by smithdc1** [comment]
@felixxm -- I had another go at this, as conda-forge now has updated packages available. It seems `Proj9` works but seems there are changes in GDAL 3.4x
 
The test failures would appear to be related. 🤔 

I see there was a GDAL bugfix release [today](https://github.com/OSGeo/gdal/releases/tag/v3.4.2), the [change log](https://github.com/OSGeo/gdal/blob/v3.4.2/gdal/NEWS.md) does mention `nodata` but it's unclear if it is related to the change we are seeing. Could be worth waiting a week or so and seeing what happens once the new release is available? 

<details>

```
(proj9) @smithdc1 ➜ /workspaces/django/tests (master ✗) $ python runtests.py gis_tests/ --parallel=1
Testing against Django installed in '/workspaces/django/django'
Found 548 test(s).
Creating test database for alias 'default'...
Creating test database for alias 'other'...
System check identified no issues (1 silenced).
..ssss................s..sssssss...sss..................s.......s.GEOS warning: Ring Self-intersection at or near point 1 0
..RTTOPO warning: Ring Self-intersection at or near point 1 0

.RTTOPO warning: Hole lies outside shell at or near point 10 0

..s..............................GEOS warning: Ring Self-intersection at or near point 1 0
GEOS warning: Ring Self-intersection at or near point 1 0
.s...s....s............ss..s.s..ss.................................................................................................F...............F..................................s..................ss................................................................................................s.s..ss...s.....ssssssssssssssssss........................ssssssss...........................................................................ssssss
======================================================================
FAIL: test_band_data (gis_tests.gdal_tests.test_raster.GDALBandTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspaces/django/tests/gis_tests/gdal_tests/test_raster.py", line 765, in test_band_data
    self.assertEqual(band.nodata_value, 15)
AssertionError: 10 != 15

======================================================================
FAIL: test_raster_info_accessor (gis_tests.gdal_tests.test_raster.GDALRasterTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspaces/django/tests/gis_tests/gdal_tests/test_raster.py", line 385, in test_raster_info_accessor
    self.assertIn(line, info_lines)
AssertionError: 'NoData Value=15' not found in ['Driver: GTiff/GeoTIFF', 'Files: /workspaces/django/tests/gis_tests/gdal_tests/../data/rasters/raster.tif', '/workspaces/django/tests/gis_tests/gdal_tests/../data/rasters/raster.tif.aux.xml', 'Size is 163, 174', 'Coordinate System is:', 'PROJCRS["NAD83 / Florida GDL Albers",', 'BASEGEOGCRS["NAD83",', 'DATUM["North American Datum 1983",', 'ELLIPSOID["GRS 1980",6378137,298.257222101,', 'LENGTHUNIT["metre",1]]],', 'PRIMEM["Greenwich",0,', 'ANGLEUNIT["degree",0.0174532925199433]],', 'ID["EPSG",4269]],', 'CONVERSION["Florida GDL Albers (meters)",', 'METHOD["Albers Equal Area",', 'ID["EPSG",9822]],', 'PARAMETER["Latitude of false origin",24,', 'ANGLEUNIT["degree",0.0174532925199433],', 'ID["EPSG",8821]],', 'PARAMETER["Longitude of false origin",-84,', 'ANGLEUNIT["degree",0.0174532925199433],', 'ID["EPSG",8822]],', 'PARAMETER["Latitude of 1st standard parallel",24,', 'ANGLEUNIT["degree",0.0174532925199433],', 'ID["EPSG",8823]],', 'PARAMETER["Latitude of 2nd standard parallel",31.5,', 'ANGLEUNIT["degree",0.0174532925199433],', 'ID["EPSG",8824]],', 'PARAMETER["Easting at false origin",400000,', 'LENGTHUNIT["metre",1],', 'ID["EPSG",8826]],', 'PARAMETER["Northing at false origin",0,', 'LENGTHUNIT["metre",1],', 'ID["EPSG",8827]]],', 'CS[Cartesian,2],', 'AXIS["easting (X)",east,', 'ORDER[1],', 'LENGTHUNIT["metre",1]],', 'AXIS["northing (Y)",north,', 'ORDER[2],', 'LENGTHUNIT["metre",1]],', 'USAGE[', 'SCOPE["State-wide spatial data management."],', 'AREA["United States (USA) - Florida."],', 'BBOX[24.41,-87.63,31.01,-79.97]],', 'ID["EPSG",3086]]', 'Data axis to CRS axis mapping: 1,2', 'Origin = (511700.468070655711927,435103.377123198588379)', 'Pixel Size = (100.000000000000000,-100.000000000000000)', 'Metadata:', 'AREA_OR_POINT=Area', 'Image Structure Metadata:', 'INTERLEAVE=BAND', 'Corner Coordinates:', 'Upper Left  (  511700.468,  435103.377) ( 82d51\'46.16"W, 27d55\' 1.53"N)', 'Lower Left  (  511700.468,  417703.377) ( 82d51\'52.04"W, 27d45\'37.50"N)', 'Upper Right (  528000.468,  435103.377) ( 82d41\'48.81"W, 27d54\'56.30"N)', 'Lower Right (  528000.468,  417703.377) ( 82d41\'55.54"W, 27d45\'32.28"N)', 'Center      (  519850.468,  426403.377) ( 82d46\'50.64"W, 27d50\'16.99"N)', 'Band 1 Block=163x50 Type=Byte, ColorInterp=Gray', 'NoData Value=10']

----------------------------------------------------------------------
Ran 548 tests in 7.394s

FAILED (failures=2, skipped=67)
Destroying test database for alias 'default'...
Destroying test database for alias 'other'...

(proj9) @smithdc1 ➜ /workspaces/django/tests (master ✗) $ conda list
# packages in environment at /home/codespace/.conda/envs/proj9:
#
# Name                    Version                   Build  Channel
...
gdal                      3.4.1           py310h8172e47_6    conda-forge
geos                      3.10.2               h9c3ff4c_0    conda-forge
...
libgdal                   3.4.1                hda961b2_6    conda-forge
...
libspatialite             5.0.1               ha867d66_15    conda-forge
...
proj                      9.0.0                h93bde94_0    conda-forge
...
```
</details>

**Comment by felixxm** [comment]
@smithdc1 I found two failing tests with SpatiaLite and GDAL 3.4.2. I'll check 3.4.1, 3.4.0, etc. to find out when they change the previous behavior and prepare tests cleanups.

**Comment by felixxm** [comment]
I added a separate ticket for GDAL 3.4+ support, see ticket-33577.

**Comment by felixxm** [comment]
> The test failures would appear to be related. :thinking: 

@smithdc1 I fixed tests in 970f5bf5035d8c99b0ff883f984b60cf531dddbe, so you can try again in PROJ 9.

**Comment by smithdc1** [comment]
Will do, it's back on my list.

**Comment by felixxm** [comment]
@smithdc1 Is it ready?

**Comment by smithdc1** [comment]
Yes. 👍

**Comment by felixxm** [comment]
Great, will check :+1:

File: docs/ref/contrib/gis/install/geolibs.txt [hunk 1]
```diff
@@ -173,12 +183,25 @@ done *prior* to configuration::
...
+
 Finally, configure, make and install PROJ::
 
-    $ ./configure
-    $ make
-    $ sudo make install
-    $ cd ..
+    $ cmake ..
+    $ cmake --build .
+    $ cmake --build . --target install
```
Comment (felixxm) [inline docs/ref/contrib/gis/install/geolibs.txt]:
```suggestion
    $ sudo cmake --build . --target install
```

File: docs/ref/contrib/gis/install/geolibs.txt [hunk 2]
```diff
@@ -173,12 +183,25 @@ done *prior* to configuration::
     $ tar xzf ../../proj-data-X.Y.tar.gz
     $ cd ..
 
+For PROJ 9.0x and greater, releases only support builds using ``CMake``
+following implementation of `Proj RFC-7`_.
```
Comment (felixxm) [inline docs/ref/contrib/gis/install/geolibs.txt]:
```suggestion
(see `PROJ RFC-7`_).
```

**Review by felixxm — COMMENTED** [review]
@smithdc1 Thanks :+1: I moved changes in GEOS installation to a separate commit, and pushed small edits.

**Pull request merged**